Sources/HiPMobile/BusinessLayer/Managers/ExhibitManager.cs

```diff
@@ -24,80 +24,73 @@ namespace PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Managers
     /// </summary>
     public static class ExhibitManager
     {
-        private static readonly IDataAccess DataAccess = IoCManager.Resolve<IDataAccess>();
+        public static ReadExtensions Exhibits(this IReadOnlyDataAccess dataAccess) => new ReadExtensions(dataAccess);
+
+        public static ReadWriteExtensions Exhibits(this ITransactionDataAccess dataAccess) => new ReadWriteExtensions(dataAccess);
 
-        /// <summary>
-        /// Gets an exhibitset with a specific id.
-        /// </summary>
-        /// <param name="id">The id of the exhibitset to be retrived.</param>
-        /// <returns>The exhibitSet with the given id. If it doesn't exist, null is returned.</returns>
-        public static ExhibitSet GetExhibitSet(string id)
+        public class ReadExtensions
         {
-            if (!string.IsNullOrEmpty(id))
+            private readonly IReadOnlyDataAccess dataAccess;
+
+            public ReadExtensions(IReadOnlyDataAccess dataAccess)
             {
-                return DataAccess.GetItem<ExhibitSet>(id);
+                this.dataAccess = dataAccess ?? throw new ArgumentNullException(nameof(dataAccess));
             }
-            return null;
-        }
 
-        /// <summary>
-        /// Get all available exhibitsets.
-        /// </summary>
-        /// <returns>The enumerable of all available exibitsets.</returns>
-        public static IEnumerable<ExhibitSet> GetExhibitSets()
-        {
-            return DataAccess.GetItems<ExhibitSet>();
-        }
+            /// <summary>
+            /// Get an exhibit with a specific id including its image and the IDs of its pages.
+            /// </summary>
+            /// <param name="id">The id of the exhibit to be retrived.</param>
+            /// <returns>The exhibit with the given id. If no exhibit exists, null is returned.</returns>
+            public Exhibit GetExhibit(string id)
+            {
+                if (!string.IsNullOrEmpty(id))
+                {
+                    return dataAccess.GetItem<Exhibit>(id,
+                        nameof(Exhibit.Image),
+                        "PagesRefs.Page.Audio");
```

Comment:
Yes, it's a tradeoff. The non-breaking alternative would be `$"{nameof(Exhibit.PagesRefs)}.{nameof(JoinExhibitPage.Page)}.{nameof(Page.Audio)}"` which is quite hard to read. But we can change it if you like.

Reply:
While it is slightly harder to read, I don't think we'd read this code very often and it would increase safety. As you say it's a tradeoff, so I'll leave the decision up to you.